src/nvim/window.c

```diff
@@ -5771,17 +5771,25 @@ void may_trigger_win_scrolled_resized(void)
 
   recursive = true;
 
+  // Save window info before autocmds since they can free windows
+  char resize_winid[NUMBUFLEN];
+  if (trigger_resize) {
+    vim_snprintf(resize_winid, sizeof(resize_winid), "%d", first_size_win->handle);
+  }
+
+  char scroll_winid[NUMBUFLEN];
+  if (trigger_scroll) {
+    vim_snprintf(scroll_winid, sizeof(scroll_winid), "%d", first_scroll_win->handle);
+  }
+
   // If both are to be triggered do WinResized first.
   if (trigger_resize) {
     save_v_event_T save_v_event;
     dict_T *v_event = get_v_event(&save_v_event);
 
     if (tv_dict_add_list(v_event, S_LEN("windows"), windows_list) == OK) {
       tv_dict_set_keys_readonly(v_event);
-
-      char winid[NUMBUFLEN];
-      vim_snprintf(winid, sizeof(winid), "%d", first_size_win->handle);
-      apply_autocmds(EVENT_WINRESIZED, winid, winid, false, first_size_win->w_buffer);
+      apply_autocmds(EVENT_WINRESIZED, resize_winid, resize_winid, false, curbuf);
```

Comment:
Using `curbuf` here, I think, changes the meaning of `<abuf>`? If we save the handles as I described, then here we can check `bufref_valid` : 

```c
buf_T *buf = bufref_valid(resize_bufref) ? resize_bufref.br_buf : curbuf;
apply_autocmds(EVENT_WINRESIZED, resize_winid, resize_winid, false, curbuf);
```

Then we still may need to update the documentation. Or maybe we should throw an error instead of calling `apply_autocmds`, if the buf is no longer valid.